omas/omas_physics.py

```diff
@@ -180,6 +181,102 @@ def summary_greenwald(ods, update=True):
+
+    Rb = ods['equilibrium']['time_slice'][time_index]['boundary']['outline']['r']
+    Zb = ods['equilibrium']['time_slice'][time_index]['boundary']['outline']['z']
+
+
+    Rgrid = ods['equilibrium']['time_slice'][time_index]['profiles_2d'][0]['grid']['dim1']
+    Zgrid = ods['equilibrium']['time_slice'][time_index]['profiles_2d'][0]['grid']['dim2']
+
+    psi2d = ods['equilibrium']['time_slice'][time_index]['profiles_2d'][0]['psi']
+    psi_interp = scipy.interpolate.interp2d(Zgrid, Rgrid, psi2d)
+    psi_eq = ods['equilibrium']['time_slice'][time_index]['profiles_1d']['psi']
+    rhon_eq = ods['equilibrium']['time_slice'][time_index]['profiles_1d']['rho_tor_norm']
+    rhon_cp = ods['core_profiles']['profiles_1d'][time_index]['grid']['rho_tor_norm']
+    ne = ods['core_profiles']['profiles_1d'][time_index]['electrons']['density_thermal']
+    ne = numpy.interp(rhon_eq, rhon_cp, ne)
+    tck = scipy.interpolate.splrep(psi_eq, ne, k=3)
+    
+    if 'time' not in ods['interferometer']:
+        ods_n['interferometer.ids_properties.homogeneous_time'] = 1
+        ods_n['interferometer']['time'] = copy.copy(ods['core_profiles']['time'])
+    
+    ifpaths = [['first_point', 'second_point'], ['second_point', 'third_point',]]
+    for channel in ods['interferometer']['channel']:
+        ne_line_paths = []
+        dist_paths = [] 
+        for ifpath in ifpaths:
+            R1 = ods['interferometer']['channel'][channel]['line_of_sight'][ifpath[0]]['r']
+            Z1 = ods['interferometer']['channel'][channel]['line_of_sight'][ifpath[0]]['z']
+            phi1 = ods['interferometer']['channel'][channel]['line_of_sight'][ifpath[0]]['phi']
+            x1 = R1*numpy.cos(phi1)
+            y1 = R1*numpy.sin(phi1)
+            		
+            R2 = ods['interferometer']['channel'][channel]['line_of_sight'][ifpath[1]]['r']
+            Z2 = ods['interferometer']['channel'][channel]['line_of_sight'][ifpath[1]]['z']
+            phi2 = ods['interferometer']['channel'][channel]['line_of_sight'][ifpath[1]]['phi']
+            x2 = R2*numpy.cos(phi2)
+            y2 = R2*numpy.sin(phi2)
+                
+            xline = (x2-x1) * numpy.linspace(0, 1, line_grid) + x1
+            yline = (y2-y1) * numpy.linspace(0, 1, line_grid) + y1			
+            Rline = (R2-R1) * numpy.linspace(0, 1, line_grid) + R1
+            Zline = (Z2-Z1) * numpy.linspace(0, 1, line_grid) + Z1
+            dist = numpy.zeros(line_grid)
+ 
+            for i, Rval in enumerate(Rline):
+                dist[i]= numpy.min((Rline[i]-Rb)**2 + (Zline[i]-Zb)**2 )
+			
+            zero_crossings = numpy.where(numpy.diff(numpy.sign(numpy.gradient(dist))))[0]
+            i1 = zero_crossings[0]
+            i2 = zero_crossings[-1]
+
+            ne_line = 0.
+            for i in range(i1, i2, numpy.sign(i2-i1)):
+                psival = psi_interp(Zline[i], Rline[i])[0]
+                ne_interp  = scipy.interpolate.splev(psival, tck)
+                ne_line  += ne_interp
```

Comment:
If you used the trapezoid rule, could you get away with fewer points along the line and thus make it faster?
```suggestion
            psival = [psi_interp(Zline[i], Rline[i])[0] for i in range(i1, i2, numpy.sign(i2-i1))]
            ne_interp  = scipy.interpolate.splev(psival, tck)
            ne_line  = numpy.trapz(ne_interp)
```